lib/iris/experimental/stratify.py

```diff
@@ -0,0 +1,179 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Routines for putting data on new strata (aka. isosurfaces), often in the
+Z direction.
+
+"""
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+from functools import partial
+
+import numpy as np
+import stratify
+
+from iris.cube import Cube
+from iris.coords import Coord, AuxCoord, DimCoord
+
+
+def _copy_coords_without_dim(cube, result, z_dim):
+    """
+    Helper function to copy across non-z coordinates between cubes.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube instance
+        Incoming cube containing the coordinates to be copied.
+    result : iris.cube.Cube instance
+        Incoming cube for the coordinates to be copied to.
+    z_dim : Dimension within the cube of the z-dimension
+        The dimension will not be copied, as the incoming z-dimension
+        is model level_height, whilst the outgoing z-dimension is pressure.
+
+    Returns
+    -------
+    Cube
+        A single cube with the non-z dimensions added to
+        incoming cube.
+
+    """
+    # Copy across non-Z coordinates.
+    for coord in cube.dim_coords:
+        dim, = cube.coord_dims(coord)
```

Comment:
@pelson Happy for you to stick with your `[dim] =` pattern, rather than `dim, =` pattern :wink: